I don't think that the CI failures are spurious. I found this in the logs of the failed `Check` job:

```
[lock_api 0.4.10] error[E0463]: can't find crate for `std`
  [lock_api 0.4.10]   |
  [lock_api 0.4.10]   = note: the `x86_64-unknown-uefi` target may not be installed
  [lock_api 0.4.10]   = help: consider downloading the target with `rustup target add x86_64-unknown-uefi`
  [lock_api 0.4.10]   = help: consider building the standard library from source with `cargo build -Zbuild-std`
  [lock_api 0.4.10] 
  [lock_api 0.4.10] error: aborting due to 1 previous error
  [lock_api 0.4.10] 
  [lock_api 0.4.10] For more information about this error, try `rustc --explain E0463`.
  [lock_api 0.4.10] error[E0463]: can't find crate for `core`
  [lock_api 0.4.10]   |
  [lock_api 0.4.10]   = note: the `x86_64-unknown-uefi` target may not be installed
  [lock_api 0.4.10]   = help: consider downloading the target with `rustup target add x86_64-unknown-uefi`
  [lock_api 0.4.10]   = help: consider building the standard library from source with `cargo build -Zbuild-std`
  [lock_api 0.4.10] 
  [lock_api 0.4.10] error: aborting due to 1 previous error
  [lock_api 0.4.10] 
  [lock_api 0.4.10] For more information about this error, try `rustc --explain E0463`.
  [lock_api 0.4.10] warning: autocfg could not probe for `std`
```

So something is trying to bring in a `std` dependency, which doesn't work for this bare-metal project.

I'll look into it, have a good vacation @phip1611!


_edit:_ Ah, that's just a `autocfg` thing, not an actual error.

--- a/uefi/src/main.rs
+++ b/uefi/src/main.rs
@@ -242,8 +242,8 @@ fn allocate_loader_data(size: usize) -> &'static mut [u8] {
         ((size - 1) / 4096) + 1,
     )
     .expect("Failed to allocate memory for the file");
-    unsafe { ptr::write_bytes(ptr.as_mut(), 0, size) };
-    unsafe { slice::from_raw_parts_mut(ptr.as_mut(), size) }
+    unsafe { ptr::write_bytes(ptr.as_ptr(), 0, size) };
+    unsafe { slice::from_raw_parts_mut(ptr.as_ptr(), size) }
 }
 
 /// Opens PXE Base Code for the device path that loaded this image.

